src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -451,7 +479,15 @@ private SyntaxTree WithChanges(SourceText newText, IReadOnlyList<TextChangeRange
             using (var parser = new InternalSyntax.LanguageParser(lexer, oldTree?.GetRoot(), changes))
             {
                 var compilationUnit = (CompilationUnitSyntax)parser.ParseCompilationUnit().CreateRed();
-                var tree = new ParsedSyntaxTree(newText, newText.Encoding, newText.ChecksumAlgorithm, this.FilePath, this.Options, compilationUnit, parser.Directives);
+                var tree = new ParsedSyntaxTree(
+                    newText,
+                    newText.Encoding,
+                    newText.ChecksumAlgorithm,
+                    this.FilePath,
+                    this.Options,
+                    compilationUnit,
+                    parser.Directives,
+                    diagnosticOptions: null);
```

Comment:
>null [](start = 39, length = 4)

This looks unexpected. It feels like we should preserve DiagnosticOptions from the original tree. We do this for parse options.